### PR TITLE
FurryR's changes

### PR DESCRIPTION
**Pull request by SimonShiki**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -49,6 +49,9 @@
         "keyword-spacing": [2, {
             "before": true,
             "after": true
-        }]
+        }],
+        // disabled rules
+        "@typescript-eslint/triple-slash-reference": "off",
+        "@typescript-eslint/no-explicit-any": "off"
```
Comment (SimonShiki) [inline .eslintrc.json]: unnecessary

File: .gitignore [hunk 2]
```diff
@@ -7,6 +7,10 @@ npm-*
 
 # Yarn
 yarn-error.log
+.yarn/
+.pnp.cjs
+.pnp.loader.mjs
+.yarnrc.yml
```
Comment (SimonShiki) [inline .gitignore]: remove

File: src/injector/inject.ts [hunk 3]
```diff
@@ -178,33 +203,31 @@ export function inject (vm: EurekaCompatibleVM) {
...
                                 : 'unsandboxed') as 'unsandboxed' | 'sandboxed'
                     );
-                    const extensionId = loader.getIdByUrl(url);
-                    // @ts-expect-error internal hack
-                    vm.extensionManager._loadedExtensions.set(extensionId, 'Eureka');
+                    /*
+                     * Const extensionId = loader.getIdByUrl(url);
+                     * @ ts-expect-error internal hack
+                     * vm.extensionManager._loadedExtensions.set(extensionId, 'Eureka');
+                     */
```
Comment (SimonShiki) [inline src/injector/inject.ts]: ~~use /**/ for disabled code statement~~
remove if it has been moved to loader

File: src/injector/inject.ts [hunk 4]
```diff
@@ -358,16 +380,39 @@ export function inject (vm: EurekaCompatibleVM) {
     }
 
     // Blockly stuffs
-    vm.once('workspaceUpdate', () => {
-        const blockly = (window.eureka.blockly = getBlocklyInstance(vm));
-        if (!blockly) {
+    let initalized = false;
+    getBlocklyInstance(vm).then((blockly) => {
+        if (!initalized) {
```
Comment (SimonShiki) [inline src/injector/inject.ts:388]: How do you make sure you actually get blockly instance here?

File: src/injector/inject.ts [hunk 5]
```diff
@@ -358,16 +380,39 @@ export function inject (vm: EurekaCompatibleVM) {
...
+                originalAddCreateButton_.call(this, workspace, xmlList, ...args);
+                injectToolbox(xmlList, workspace, format);
+            };
+            const workspace = blockly.getMainWorkspace();
+            workspace.getToolbox().refreshSelection();
+            workspace.toolboxRefreshEnabled_ = true;
+        }
+    });
+    /*
+     * Const blockly = (window.eureka.blockly = getBlocklyInstance(vm));
```
Comment (SimonShiki) [inline src/injector/inject.ts]: use ``/**/`` for disabled code statements

File: src/injector/inject.ts [hunk 6]
```diff
@@ -49,38 +48,64 @@ function getExtensionIdForOpcode (opcode: string) {
...
-    function hijack (fn: (...args: unknown[]) => unknown) {
-        const _orig = Function.prototype.apply;
-        Function.prototype.apply = function (thisArg: any) {
-            return thisArg;
-        };
-        const result = fn();
-        Function.prototype.apply = _orig;
-        return result;
-    }
+async function getBlocklyInstance (vm: EurekaCompatibleVM): Promise<any> {
```
Comment (SimonShiki) [inline src/injector/inject.ts:52]: where is timeout?

File: src/loader/make-ctx.ts [hunk 7]
```diff
@@ -69,7 +69,7 @@ function createTranslate (vm?: VM) {
     };
 
     let storedTranslations = {};
-    translate.setup = (newTranslations: Message | {} | null) => {
+    translate.setup = (newTranslations: Message | object | null) => {
```
Comment (SimonShiki) [inline src/loader/make-ctx.ts:72]: does it make any difference?

File: src/util/settings.ts [hunk 8]
```diff
@@ -28,33 +28,35 @@ export function getSettingsFromStorage () {
...
-        window.localStorage.setItem(SETTINGS_KEY, JSON.stringify(obj));
-    } catch (_: unknown) {
-        const newObject = Object.assign({}, puppet, {
-            [prop]: value
-        });
-        window.localStorage.setItem(SETTINGS_KEY, JSON.stringify(newObject));
-    }
-}
+/*
+ * Function saveSettingsToStorage (prop: string, value: string) {
```
Comment (SimonShiki) [inline src/util/settings.ts]: remove if unnecessary

File: src/injector/inject.ts (hunk 4 above)
Comment (FurryR) [reply src/injector/inject.ts:388]: If `EXTENSION_ADDED` event does not exist or has no `ScratchBlocks`, the promise will not resolve.

File: src/loader/make-ctx.ts (hunk 7 above)
Comment (FurryR) [reply src/loader/make-ctx.ts:72]: eslint: avoid `{}` where possible -- `{}` actually means `any non-nullish value`.

File: src/injector/inject.ts (hunk 6 above)
Comment (FurryR) [reply src/injector/inject.ts:52]: see `setTimeout` below

Comment (SimonShiki) [reply src/injector/inject.ts:52]: timeout is not only necessary for alternative method, but also for this

Comment (FurryR) [reply src/injector/inject.ts:52]: `initalized` flag should work. If timeout exceeded, `initalized` flag will be set to `true` and then `getBlocklyInstance` callback will be aborted.

**Pull request merged**